internal/resource/http/http_test.go

```diff
@@ -0,0 +1,295 @@
+package http
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	resourceSvc "github.com/hashicorp/consul/agent/grpc-external/services/resource"
+	svctest "github.com/hashicorp/consul/agent/grpc-external/services/resource/testing"
+	pbdemov1 "github.com/hashicorp/consul/proto/private/pbdemo/v1"
+
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/internal/resource/demo"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	pbdemov2 "github.com/hashicorp/consul/proto/private/pbdemo/v2"
+	"github.com/hashicorp/consul/sdk/testutil"
+)
+
+const testACLTokenArtistReadPolicy = "00000000-0000-0000-0000-000000000001"
+const testACLTokenArtistWritePolicy = "00000000-0000-0000-0000-000000000002"
+
+func parseToken(req *http.Request, token *string) {
+	*token = req.Header.Get("x-Consul-Token")
```

Comment:
nit: might want to stick to one of the two existing widely used versions of this verbatim string for consistency instead of creating a new variation that is a bit different in case (yes, even though headers are case in-sensetive).

```
$ git grep "x-consul-token" | wc -l                                                                                                                                                                                                                                              
63

$ git grep "X-Consul-Token" | wc -l                                                                                                                                                                                                                                             
309

$ git grep "x-Consul-Token" | wc -l                                                                                                                                                                                                                                               
1
```

Reply:
nice catch!